lib/directory-view.coffee

```diff
@@ -106,18 +113,41 @@ class DirectoryView extends HTMLElement
     if @isExpanded then @collapse(isRecursive) else @expand(isRecursive)
 
   expand: (isRecursive=false) ->
+    initial = Promise.resolve(true)
     unless @isExpanded
       @isExpanded = true
       @classList.add('expanded')
       @classList.remove('collapsed')
-      @directory.expand()
+      initial = initial.then => @directory.expand()
 
     if isRecursive
-      for entry in @entries.children when entry instanceof DirectoryView
-        entry.expand(true)
-
+      children = (entry for entry in @entries.children when entry instanceof DirectoryView)
```

Comment:
I'm not understanding why you're special casing on the existence of `firstEntry`... Is it to check for the completion the view's directory's expansion? If so could you just always wait on the promise, which might already be resolved and do work immediately?